SU2_CFD/src/solvers/CNEMOEulerSolver.cpp

```diff
@@ -815,45 +815,71 @@ void CNEMOEulerSolver::Source_Residual(CGeometry *geometry, CSolver **solver_con
 
   unsigned short iVar, jVar;
   unsigned long iPoint;
-  unsigned long eAxi_local, eChm_local, eVib_local;
-  unsigned long eAxi_global, eChm_global, eVib_global;
 
   /*--- Assign booleans ---*/
   bool err        = false;
   bool implicit   = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
   bool frozen     = config->GetFrozen();
   bool monoatomic = config->GetMonoatomic();
+  bool axisymm    = config->GetAxisymmetric();
   bool viscous    = config->GetViscous();
   bool rans       = (config->GetKind_Turb_Model() != NONE);
 
   CNumerics* numerics = numerics_container[SOURCE_FIRST_TERM];
 
   /*--- Initialize the error counter ---*/
-  eAxi_local = 0;
-  eChm_local = 0;
-  eVib_local = 0;
+  unsigned long eAxi_local = 0;
+  unsigned long eChm_local = 0;
+  unsigned long eVib_local = 0;
 
   /*--- Initialize the source residual to zero ---*/
   for (iVar = 0; iVar < nVar; iVar++) Residual[iVar] = 0.0;
```

Comment:
That is unlikely, the NEMO solver does not declare itself to be OpenMP-ready (via method GetHasHybridParallel() -> true) and so it will not be called multi-threaded by the Integration classes.
Also if you do not compile with -Dwith-omp=enabled (or true, I can never recall) all those SU2_OMP_? macros are no-ops.

Reply:
So this was a super complex problem to solve....
I had the gradient computation inside the for loop.  So as you can imagine, things went south quite quickly.

Reply:
Actually you did point me to an hybrid parallel bug...